python/lsst/afw/geom/testUtils.py

```diff
@@ -91,3 +100,759 @@ def __iter__(self):
+        fromAxes, toAxes : integer
+            number of axes in fromFrame and toFrame, respectively
+        """
+        transformClassName = "Transform{}To{}".format(fromName, toName)
+        transformClass = getattr(afwGeom, transformClassName)
+        baseMsg = "transformClass={}".format(transformClass.__name__)
+        for nIn in self.goodNaxes[fromName]:
+            # check invalid numbers of output for the given toName
+            for badNout in self.badNaxes[toName]:
+                badPolyMap = makeTwoWayPolyMap(nIn, badNout)
+                msg = "{}, nIn={}, badNout={}".format(baseMsg, nIn, badNout)
+                with self.assertRaises(InvalidParameterError, msg=msg):
+                    transformClass(badPolyMap)
+
+            # check valid numbers of outputs for the given toName
+            for nOut in self.goodNaxes[toName]:
+                msg = "{}, nIn={}, nOut={}".format(baseMsg, nIn, nOut)
+                polyMap = makeTwoWayPolyMap(nIn, nOut)
+                transform = transformClass(polyMap)
+
+                desStr = "{}[{}->{}]".format(transformClassName, nIn, nOut)
+                self.assertEqual("{}".format(transform), desStr)
+                self.assertEqual(repr(transform), "lsst.afw.geom." + desStr)
+
+                self.checkTransformation(transform, polyMap, msg=msg)
+
+                # Forward transform but no inverse
+                polyMap = makeForwardPolyMap(nIn, nOut)
+                transform = transformClass(polyMap)
+                self.checkTransformation(transform, polyMap, msg=msg)
+
+                # Inverse transform but no forward
+                polyMap = makeForwardPolyMap(nOut, nIn).getInverse()
+                transform = transformClass(polyMap)
+                self.checkTransformation(transform, polyMap, msg=msg)
+
+        # check invalid numbers of inputs with valid and invalid #s of inputs
+        for badNin in self.badNaxes[fromName]:
+            for nOut in self.goodNaxes[toName] + self.badNaxes[toName]:
+                badPolyMap = makeTwoWayPolyMap(badNin, nOut)
+                msg = "{}, badNin={}, nOut={}".format(baseMsg, nIn, nOut)
+                with self.assertRaises(InvalidParameterError, msg=msg):
+                    transformClass(badPolyMap)
+
+    def checkTransformFromFrameSet(self, fromName, toName):
+        transformClassName = "Transform{}To{}".format(fromName, toName)
+        transformClass = getattr(afwGeom, transformClassName)
+        baseMsg = "transformClass={}".format(transformClass.__name__)
+        for nIn in self.goodNaxes[fromName]:
+            for nOut in self.goodNaxes[toName]:
+                msg = "{}, nIn={}, nOut={}".format(baseMsg, nIn, nOut)
+
+                baseFrame = self.makeGoodFrame(fromName, nIn)
+                currFrame = self.makeGoodFrame(toName, nOut)
+                frameSet = self.makeFrameSet(baseFrame, currFrame)
+                self.assertEqual(frameSet.getNframe(), 4)
```

Comment:
I agree the name isn't wonderful, but I have chosen to expose AST conventions to avoid nuisance differences.